src/storage/src/keyspace.rs

```diff
@@ -111,7 +144,25 @@ impl<S: StateStore> Keyspace<S> {
     /// Gets from the keyspace with the `prefixed_key` of given key.
     /// The returned value is based on a snapshot corresponding to the given `epoch`
     pub async fn get(&self, key: impl AsRef<[u8]>, epoch: u64) -> StorageResult<Option<Bytes>> {
-        self.store.get(&self.prefixed_key(key), epoch).await
+        self.store.get(&self.prefixed_key(key), epoch, None).await
+    }
+
+    pub async fn get_with_vnode(
+        &self,
+        key: impl AsRef<[u8]>,
+        epoch: u64,
+        vnode: VirtualNode,
+    ) -> StorageResult<Option<Bytes>> {
+        // Construct vnode bitmap.
+        let mut bitmap_inner = [0; VNODE_BITMAP_LEN];
```

Comment:
This code seems to appear in multiple places. Is it possible to have a `VNodeBitmap::new(vnode, table_id)`, let the caller to provide a VNodeBitmap?

Reply:
`VNodeBitmap` is actually a proto type. Maybe we should define a non-proto type for it.